core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -141,11 +124,12 @@ private[log] class ProducerStateEntry(val producerId: Long,
   }
 
   def update(nextEntry: ProducerStateEntry): Unit = {
-    maybeUpdateEpoch(nextEntry.producerEpoch)
+    maybeUpdateProducerEpoch(nextEntry.producerEpoch)
```

Comment:
This maybe out of the scope of this PR but I found the call trace of `addBatch` and `update` are basically the same: the latter is from recover, append, and load, and the former is also from append and load, can we consolidate them into a single function then, or are there any scenarios that would only get on but not the other?

Reply:
Yeah, it's a fair point. I looked into this, but found it not very straightforward. If it's ok, let's do this in a follow-up.

Reply:
SG